### PR TITLE
Fix/47308 workspace feed cleanup

### PR DESCRIPTION
**Pull request by koko57**
### Fixed Issues

$ https://github.com/Expensify/App/issues/47308
$ https://github.com/Expensify/App/issues/48376
PROPOSAL:

### Tests

PREREQUISITES: workspace feed / all betas enabled, Expensify Card enabled, accounting enabled

Reconciliation Account 

1. Go to Workspaces -> Accounting -> Card Reconciliation -> Reconciliation account
2. Change account
3. Verify that the UpdateExpensifyCardSettlementAccount request is sent 

Card Name 

1. Go to Issue New Card flow
2. Go through the flow, do not change the default card name (should be "[userName]'s Card")
3. Go to Confirmation, then select Cardholder to change the assignee
4. Change the assignee
5. After going back to confirmation the name should be "[userName]'s Card" where userName will change to the name of the newly selected user
6. (You can do it in the same flow before you issue the card or in a new flow) Go to the card name step, change it for a custom name.
7. Repeat steps 3-4.
8. Verify that after changing the assignee the custom name is not changed.

- [x] Verify that no errors appear in the JS console

### QA Steps

same as above

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
        - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/dbe3dd6f-4613-4dd4-b2ac-907dc281e217

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/cd67faff-9c0b-4f3b-9aeb-0dfaa5ed94b0

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/8e3fb8a2-929b-43f7-99fb-eeed1018611e

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/fd67d1fd-7ab2-4f08-a245-62d74ee33740

Screenshot 2024-08-30 at 17 34 06

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/8652c880-7440-40cd-b2dd-d2e8a7dd1d78

</details>

### CODE REVIEW

**Comment by koko57** [comment]
@mountiny I'm getting code 666 for the response, but it's probably because I don't have accounting connection set up and I just did a workaround to test it.

**Comment by mountiny** [comment]
The react compiler action is not happy.

You can create a free trial qbo account to connect to qbo and test that i think. @allgandalf or @DylanDylann  could try that

**Comment by koko57** [comment]
Strange, it failed to compile a lot components that I haven't even touched

Screenshot 2024-09-02 at 18 59 24

**Comment by mountiny** [comment]
I think the action takes a diff of files it can compile on main an on thi branch and if there is a single file that is newly un-compilable it would fail. So its ok it is listing files we did not touch but there will be probably one that this PR made unable to compile. I will rerun this action just in case

File: src/pages/workspace/expensifyCard/WorkspaceCardsListLabel.tsx [hunk 1]
```diff
@@ -50,9 +52,13 @@ function WorkspaceCardsListLabel({type, value, style}: WorkspaceCardsListLabelPr
     const [anchorPosition, setAnchorPosition] = useState({top: 0, left: 0});
     const anchorRef = useRef(null);
 
+    const workspaceAccountID = PolicyUtils.getWorkspaceAccountID(route.params.policyID);
     const policyCurrency = useMemo(() => policy?.outputCurrency ?? CONST.CURRENCY.USD, [policy]);
-    // TODO: instead of the first bankAccount on the list get settlementBankAccountID from the private_expensifyCardSettings NVP and check if that is connected via Plaid.
-    const isConnectedWithPlaid = useMemo(() => !!Object.values(bankAccountList ?? {})[0]?.accountData?.additionalData?.plaidAccountID, [bankAccountList]);
+    const [cardSettings] = useOnyx(`${ONYXKEYS.COLLECTION.PRIVATE_EXPENSIFY_CARD_SETTINGS}${workspaceAccountID}`);
+    const isConnectedWithPlaid = useMemo(
```
Comment (mountiny) [inline src/pages/workspace/expensifyCard/WorkspaceCardsListLabel.tsx:58]: I believe its complaining about this file

**Comment by koko57** [comment]
@mountiny now it succeed

**Review by allgandalf — APPROVED** [review]
LGTM, lets ship it 🚀

File: src/libs/API/parameters/RequestExpensifyCardLimitIncreaseParams.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 type RequestExpensifyCardLimitIncreaseParams = {
     authToken: string | null | undefined;
-    settlementBankAccountID: string;
+    settlementBankAccountID: number;
```
[the remark this answers is not in the document]
Comment (mountiny) [reply src/libs/API/parameters/RequestExpensifyCardLimitIncreaseParams.ts:3]: Yeah this is an integer so this is fine

**Review by mountiny — APPROVED** [review]
Thanks, looks good to me, I will remove the report action test steps from the QA as it cannot be QAed yet

**Comment by mvtglobally** [comment]
@mountiny we're not able to change the reconciliation account yet, so will not be able to test that part. Any chance you can QA this PR internally to ensure full coverage?

**Comment by roryabraham** [comment]
@mountiny I'm checking this off as the last PR on the checklist despite the fact that Applause hasn't been able to fully verify it. Please perform internal QA on this to make sure it worked as expected. Thanks!

**Pull request merged**